apps/explorer/src/hooks/useEndOfEpochTransactionFromCheckpoint.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useIotaClient } from '@iota/dapp-kit';
+import { type IotaTransactionBlockResponse } from '@iota/iota-sdk/client';
+import { useQuery } from '@tanstack/react-query';
+
+const MAX_TRANSACTIONS_PER_REQ = 50;
+
+export function useEndOfEpochTransactionFromCheckpoint(
+    checkpoinId?: string,
+    limit = MAX_TRANSACTIONS_PER_REQ,
+) {
+    const client = useIotaClient();
+    return useQuery({
+        queryKey: ['end-of-epoch-transaction', checkpoinId],
+        queryFn: async () => {
+            let cursor: string | undefined | null = null;
+            const filter = checkpoinId ? { Checkpoint: checkpoinId } : undefined;
+            // keep fetching until cursor is null or undefined
+            do {
+                const { data: transactions, nextCursor } = await client.queryTransactionBlocks({
+                    filter,
```

Comment:
Couldn't you pass a filter searching for a tx with the tx kind of `EndOfEpochTransaction`? Querying all the txs just to get one doesnt seem every efficient right?

Reply:
I've looked at that but there isn't an `And` filter for the queryTransactionBlock. 
And fetching just `EndOfEpochTransaction` could end up  returning a lot of transactions.